commons-hpcc/src/main/java/org/hpccsystems/commons/benchmarking/BenchmarkResult.java

```diff
@@ -156,11 +229,14 @@ public JSONArray toFlatJson(String[] selectedResults)
 
         if (selectedResults == null || selectedResults.length == 0)
         {
-            for (HashMap.Entry<String, AveragedMetric> entry : metrics.entrySet())
+            for (HashMap.Entry<String, MetricInfo> entry : metrics.entrySet())
             {
-                AveragedMetric metric = entry.getValue();
-                metric.discardOutliers();
-                
+                MetricInfo metric = entry.getValue();
+                if (metric.metric == null)
```

Comment:
if (metric || metric.metric == null) as above?
Also, mostly for aesthetics, but if we're going to make changes in this code, might as well refactor the metric var on lines 205,234 and 248 to metricinfo or currmetric